### ISSUE TITLE
$security.createJWT: Add custom headers

### ISSUE THREAD

**Issue by obrunsmann** [pocketbase/pocketbase#5808]
I need to add a custom header to JWT. Currently only payload is possible.

**Comment by obrunsmann** [comment]
Thank you for your response, although I must admit I was a bit taken aback by the tone. I'm certainly not looking to spam the issues tracker and only posted here because I believe this functionality would be beneficial for many users.

To illustrate the necessity, consider the App Store Connect API, which requires specific JWT headers, including alg and kid. Without the ability to set custom headers directly in the JWT helper, it becomes challenging—if not impossible—to properly authenticate requests to the API. This is a standard requirement for many APIs that leverage JWT, where headers are part of the expected format for security and compatibility.

Additionally, if a JWT helper is included, it would make sense to support multiple algorithms, such as ES256, which is commonly used alongside RS256 in many authentication flows today. Flexibility in header customization and algorithm support would significantly broaden the utility of the helper across different APIs and use cases.

Thank you for considering these points, and please let me know if there’s a better place for discussions like this.

Example Docs: https://developer.apple.com/documentation/appstoreconnectapi/generating-tokens-for-api-requests

**Comment by obrunsmann** [comment]
@ganigeorgiev Thank you for your detailed answer. I understand your points, especially because I have also managed some open source projects myself and therefore know the demands of many developers.

I didn't think about the fact that the functions are not implemented in Pocketbase at all, but only passed through. I have already started to implement the necessary functions directly in golang and understand that it is disproportionate to add more functions here.

Thanks for your advice on the next version!
